jupyter_notebook_config.py

```diff
@@ -76,3 +78,49 @@
             sys.exit(155)
     else:
         print('Tags variable is not an array.', file=sys.stderr)
+
+def saveFile(file_path, token):
+    """
+    Construct a requests POST call with args and kwargs and process the
+    results.
+    Args:
+        file_path: The relative path to the file from the datadir, including filename and extension
+        token: keboola storage api token
+    Returns:
+        body: Response body parsed from json.
+    Raises:
+        requests.HTTPError: If the API request fails.
+    """
+
+    url = 'http://data-loader-api/data-loader-api/save'
+    headers = {'X-StorageApi-Token': token, 'User-Agent': 'Keboola Sandbox Autosave Request'}
+    payload = {}
+    payload['file'] = {'source': file_path, 'tags': ['autosave']}
+
+    r = requests.post(url, json=payload, headers=headers, timeout=240)
+    try:
+        r.raise_for_status()
+    except requests.HTTPError:
+        # Handle different error codes
+        raise
+    else:
+        return r.json()
+
+def script_post_save(model, os_path, contents_manager, **kwargs):
+    """
+    saves the ipynb file to keboola storage on every save within the notebook
+    """
+    if model['type'] != 'notebook':
+        return
+    log = contents_manager.log
```

Comment:
i have no idea what this is :)

Reply:
it's just a logger :-).  here's an example of the output

![image](https://user-images.githubusercontent.com/569139/73547338-fe1a0280-443e-11ea-8341-7124ef2e01b9.png)

Reply:
cool!, the name looks so weirdly random